models/protocols/InMemoryProtocol.cfc

```diff
@@ -0,0 +1,44 @@
+component extends="cbmailservices.models.AbstractProtocol" singleton accessors="true" {
+
+	property name="mail" type="array";
+
+	/**
+	 * Initialize the InMemory protocol
+	 *
+	 * @properties A map of configuration properties for the protocol
+	 */
+	public InMemoryProtocol function init( struct properties = {} ) {
+		super.init( argumentCollection = arguments );
+		variables.mail = [];
+		return this;
+	}
+
+	/**
+	 * Log the message as sent.
+	 *
+	 * @payload The payload to deliver
+	 */
+	public struct function send( required cbmailservices.models.Mail payload ) {
+		variables.mail.append( arguments.payload.getMemento() );
+		return { "error": false, "errorArray": [] };
+	}
+
+    /**
+     * Check if a given message has been sent by passing in a callback.
+     * Each message will be checked against the callback.
+     * If one message passes the callback, this method will return true.
+     * 
+     * @callback A callback function to check against each mail item.
+     */
+	public boolean function hasMessage( required function callback ) {
+		return arraySome( variables.mail, arguments.callback );
```

Comment:
@elpete unfortunately we have to support ACF2016 and `arraySome` is not supported in 2016.  We will need an alternative: Combine `arrayFilter().len()`

Reply:
Swapped and added a test for this method.